Modules/vc-module-orders/VirtoCommerce.OrdersModule.Web/Security/OrderReadPricesPermission.cs

```diff
@@ -7,23 +10,28 @@ public static class OrderReadPricesPermission
         public static string ApplyResponseGroupFiltering(Permission[] permissions, string respGroup)
         {
             var result = respGroup;
-            //TODO
-            //var needRestrict = permissions.Any() && !permissions.Any(x => x.Id == ModuleConstants.Security.Permissions.ReadPrices);
 
-            //if (needRestrict && string.IsNullOrWhiteSpace(respGroup))
-            //{
-            //    const CustomerOrderResponseGroup val = CustomerOrderResponseGroup.Full & ~CustomerOrderResponseGroup.WithPrices;
+            var needRestrict = permissions.Any() && !permissions.Any(x => x.Id == ModuleConstants.Security.Permissions.ReadPrices);
 
-            //    result = val.ToString();
-            //}
-            //else if (needRestrict)
-            //{
-            //    var items = respGroup.Split(',').Select(x => x.Trim()).ToList();
+            if (needRestrict && string.IsNullOrWhiteSpace(respGroup))
+            {
+                const CustomerOrderResponseGroup val = CustomerOrderResponseGroup.Full & ~CustomerOrderResponseGroup.WithPrices;
 
-            //    items.Remove(CustomerOrderResponseGroup.WithPrices.ToString());
+                result = val.ToString();
+            }
+            else if (needRestrict)
+            {
+                var items = respGroup.Split(',').Select(x => x.Trim()).ToList();
 
-            //    result = string.Join(",", items);
-            //}
+                items.Remove(CustomerOrderResponseGroup.WithPrices.ToString());
+
+                result = string.Join(",", items);
+            }
+
+            if (needRestrict && string.IsNullOrEmpty(result))
```

Comment:
Are you sure we need this? Doesn't it brake access permissions logic?

Reply:
I think it's ok. This code has been moved from V2. If we don't set 'default' for empty ResponseGroup   we would fill it in [this place](https://github.com/VirtoCommerce/vc-platform-core/blob/652ba31a7b0b06731052e4a0be5430026ab37ac6/Modules/vc-module-orders/VirtoCommerce.OrdersModule.Data/Services/CustomerOrderServiceImpl.cs#L71) with 'Full' value and prices wouldn't be restricted

Reply:
Correct, it was fixed in later commit that way:
https://github.com/VirtoCommerce/vc-module-order/blob/cafc166d660e1d9afda890109bba89fa54d2f33b/VirtoCommerce.OrderModule.Web/Security/OrderReadPricesPermission.cs#L30.